src/skagent/bellman.py

```diff
@@ -12,88 +12,61 @@
 
-        return {sym: post[sym] for sym in state_syms}
+    def get_arrival_states(self, calibration):
+        return self.block.get_arrival_states(calibration if calibration else self.calibration)
 
-    return transition_function
+    def get_controls(self):
+        return self.block.get_controls()
 
+    def get_shocks(self):
+        return self.block.get_shocks()
 
-def create_decision_function(block, decision_rules):
-    """
-    block
-    decision_rules
-    """
+    def transition_function(self, states_t, shocks_t, controls_t, parameters = None, decision_rules = None):
+        decision_rules = decision_rules if decision_rules else (self.decision_rules if self.decision_rules else {})
+        parameters = parameters if parameters else (self.calibration if self.calibration else {})
+
+        vals = parameters | states_t | shocks_t | controls_t
+        post = self.block.transition(vals, decision_rules, fix=list(controls_t.keys()))
+
+        return {sym: post[sym] for sym in self.arrival_states}
+
+    def decision_function(self, states_t, shocks_t, parameters = None, decision_rules = None):
+        decision_rules = decision_rules if decision_rules else (self.decision_rules if self.decision_rules else {})
+        parameters = parameters if parameters else (self.calibration if self.calibration else {})
 
-    def decision_function(states_t, shocks_t, parameters):
-        if parameters is None:
-            parameters = {}
         vals = parameters | states_t | shocks_t
-        post = block.transition(vals, decision_rules)
+        post = self.block.transition(vals, decision_rules)
         return {sym: post[sym] for sym in decision_rules}
 
-    return decision_function
 
+    def reward_function(self, states_t, shocks_t, controls_t, parameters = None, agent = None, decision_rules=None):
+        decision_rules = decision_rules if decision_rules else (self.decision_rules if self.decision_rules else {})
+        parameters = parameters if parameters else (self.calibration if self.calibration else {})
 
-def create_reward_function(block, agent=None, decision_rules=None):
-    """
-    block
-    agent : optional, str
-    decision_rules : optional, dict
-        Decision rules of control variables that will _not_
-        be given to the function.
-    """
-    decision_rules = {} if decision_rules is None else decision_rules
-
-    def reward_function(states_t, shocks_t, controls_t, parameters):
         vals_t = parameters | states_t | shocks_t | controls_t
-        post = block.transition(vals_t, decision_rules, fix=list(controls_t.keys()))
+        post = self.block.transition(vals_t, decision_rules, fix=list(controls_t.keys()))
         return {
             sym: post[sym]
-            for sym in block.reward
-            if agent is None or block.reward[sym] == agent
+            for sym in self.block.reward
+            if agent is None or self.block.reward[sym] == agent
```

Comment:
Should a BellmanPeriod only have one agent? the decision maker? since BellmanPeriod represents a bellman problem with only one decision maker, I assume